enhancements/microshift/kubernetes-for-industrial-edge.md

```diff
@@ -0,0 +1,436 @@
+   a good state if an upgrade fails.
+6. As an device administrator, I have control over the operating
+   system running under MicroShift.
+
+### Goals
+
+* Describe an approach to providing a version of Kubernetes compatible
+  with the OpenShift APIs that it includes and that meets the
+  constraints described in the Motivation section.
+* Identify the minimal set of APIs that will be included and the
+  components needed to support them.
+* Design a system that can work autonomously, without requiring
+  external orchestration tools for deployment.
+* Treat MicroShift as an application running on RHEL, with minimal
+  privileges.
+* Design a system that can be configured by someone familiar with
+  Linux, without Kubernetes expertise.
+* Follow the [MicroShift design
+  principles](https://github.com/openshift/microshift/blob/main/docs/design.md)
+
+### Non-Goals
+
+* We are not trying to fit all of OpenShift into the target hardware
+  platforms.
+* We are not trying to make MicroShift self-managing, so no core
+  OpenShift operators will be included.
+* We do not plan to support multi-node clusters.
+* The details of the default CSI driver are left to a future
+  enhancement.
+* The details of the default CNI driver are left to a future
+  enhancement.
+* Use cases beyond industrial applications, such as developer
+  environments, are out of scope for now.
+* The design of MicroShift relies heavily on features of RHEL for Edge
+  and rpm-ostree. Those features are referenced here, but their
+  implementation is not described in detail.
+
+## Proposal
+
+To meet the resource consumption constraints (RAM and CPU), MicroShift
+is designed as an all-in-one binary for the components that are
+required to launch a kubernetes control plane, combined with
+components running in containers to provide runtime services after the
+control plane is up.
+
+Today the all-in-one binary includes:
+
+* etcd
+* kubernetes API server
+* kubernetes controller manager
+* kubernetes scheduler
+* OpenShift API server
+* OpenShift controller manager
+* OpenShift SCC manager
+
+The pod-based components include:
+
+* CoreDNS
+* OpenShift HAproxy router
+* service CA operator
+* CSI driver
+* CNI driver
+
+To allow deployers to customize and manage the OS, MicroShift will run
+on [RHEL for
+Edge](https://www.redhat.com/en/resources/meet-workload-demands-edge-computing-datasheet),
+which is designed for field-deployed devices and use cases.
+MicroShift will take advantage of the existing RHEL for Edge work by
+running as an application managed by systemd and using as few
+privileges as possible. Deployers will use standard RHEL for Edge
+tools such as RPM, rpm-ostree, and Image Builder to construct system
+images and distribute software updates. Deployers will be responsible
+for configuring the operating system components and anything that runs
+on the device outside of MicroShift.
+
+Many of the OpenShift APIs are not relevant for edge device use
```

Comment:
We may want to mention why they are not relevant, e.g.
* MicroShift is meant as pure runtime, not build/dev-cluster (thus we prune Builds, BuildConfigs, etc.).
* MicroShift is not an interactive, multi-user environment (thus we prune users, projects, etc.).
* MicroShift doesn't manage OS or infrastructure (thus we prune related APIs).
* MicroShift is a monolith (thus we prune cluster orchestration-related APIs).

Reply:
added